src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -503,23 +624,33 @@ private function createNewInvoice(Order $order, Cart $cart, $isFreeShipping)
     /**
      * @param int $orderInvoiceId
      * @param Cart $cart
+     * @param array $newProducts
      *
      * @return OrderInvoice
      */
-    private function updateExistingInvoice($orderInvoiceId, Cart $cart)
+    private function updateExistingInvoice($orderInvoiceId, Cart $cart, array $newProducts)
     {
+        $precision = $this->getPrecisionFromCart($cart);
         $invoice = new OrderInvoice($orderInvoiceId);
 
         $invoice->total_paid_tax_excl += Tools::ps_round(
-            (float) $cart->getOrderTotal(false, Cart::BOTH_WITHOUT_SHIPPING),
-            2
+            (float) $cart->getOrderTotal(false, Cart::BOTH_WITHOUT_SHIPPING, $newProducts),
+            $precision
         );
         $invoice->total_paid_tax_incl += Tools::ps_round(
-            (float) $cart->getOrderTotal(true, Cart::BOTH_WITHOUT_SHIPPING),
-            2
+            (float) $cart->getOrderTotal(true, Cart::BOTH_WITHOUT_SHIPPING, $newProducts),
+            $precision
+        );
+        $invoice->total_products += (float) $cart->getOrderTotal(
+            false,
+            Cart::ONLY_PRODUCTS,
+            $newProducts
+        );
+        $invoice->total_products_wt += (float) $cart->getOrderTotal(
+            true,
+            Cart::ONLY_PRODUCTS,
+            $newProducts
```

Comment:
No need to update `$invoice->total_wrapping` like in `createInvoice` here?

Reply:
I don't know... but not the cas in legacy :tongue: